src/components/DarkModeToggle.tsx

```diff
@@ -0,0 +1,35 @@
+import useDarkMode from "use-dark-mode";
+
+/**
+ * This component should only be called on the client as it accesses window in it's body
+ */
+export default function DarkModeToggle() {
+  const darkMode = useDarkMode(
+    // @ts-ignore use-dark-mode is passing this value directly to useState, so the types are actually wrong here it should accept a lazy initial state
```

Comment:
I'm gonna submit a PR to `use-dark-mode` to update the types, hopefully they accept it.

Reply:
For prosperity: https://github.com/donavon/use-dark-mode/pull/75